duration-parser/src/test/java/DurationParserFormatterTest.java

```diff
@@ -0,0 +1,59 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.Duration;
+
+public class DurationParserFormatterTest {
+
+    DurationParserFormatter dpf = DurationParserFormatter.getInstance();
+
+    @Test
+    public void illegalArgumentExceptionTest() {
+        try {
+            dpf.parse(null);
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(true);
+            Assert.assertEquals("Date must not be null", e.getMessage());
+        }
+        try {
+            dpf.parse("");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(true);
+            Assert.assertEquals("Date must not be empty", e.getMessage());
+        }
+        try {
+            dpf.parse("1");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(true);
+            Assert.assertEquals("1 - invalid date and time format", e.getMessage());
+        }
+        try {
+            dpf.parse("d");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(true);
+            Assert.assertEquals("d - invalid date and time format", e.getMessage());
+        }
+        try {
+            dpf.parse("1d50m2w");
```

Comment:
it will throw exception (we use units only from unitFactorMap)
```
if (!unitFactorMap.containsKey(unit)) {
                throw new IllegalArgumentException(durationStr + " - invalid duration format");
            }
```

Reply:
and one more case: `w10d`

Reply:
right, I didnt take these cases into account((
fixed it